### PR TITLE
(RDCC-1041) Fix add/delete role api response for user not found or not active

### PR DESCRIPTION
**Pull request by shreedhar-hmcts**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/RDCC-1041

### Change description ###
Fix add/delete role api response for user not fount or not active

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

File: src/test/java/uk/gov/hmcts/reform/professionalapi/service/impl/ProfessionalUserServiceImplTest.java [hunk 1]
```diff
@@ -630,4 +579,33 @@ private RoleAdditionResponse createAddRoleResponse(HttpStatus status, String mes
...
+
+    }
+
+    @Test
+    public void modify_user_roles_feign_error_with_no_400_status() throws Exception {
+
+        when(feignExceptionMock.status()).thenReturn(400);
+        callModifyRolesForUser(HttpStatus.BAD_REQUEST);
+        verify(feignExceptionMock, times(2)).status();
+
```
Comment (KOTLAPRA23) [inline src/test/java/uk/gov/hmcts/reform/professionalapi/service/impl/ProfessionalUserServiceImplTest.java:598]: can you please prefix test for all the test cases

File: src/test/java/uk/gov/hmcts/reform/professionalapi/service/impl/ProfessionalUserServiceImplTest.java [hunk 2]
```diff
@@ -630,4 +579,33 @@ private RoleAdditionResponse createAddRoleResponse(HttpStatus status, String mes
...
+    }
+
+    public void callModifyRolesForUser(HttpStatus status) {
+        when(userProfileFeignClient.modifyUserRoles(any(), any(), any())).thenThrow(feignExceptionMock);
```
Comment (KOTLAPRA23) [inline src/test/java/uk/gov/hmcts/reform/professionalapi/service/impl/ProfessionalUserServiceImplTest.java:602]: add annotation @Test here

File: src/test/java/uk/gov/hmcts/reform/professionalapi/controller/internal/ProfessionalUserInternalControllerTest.java [hunk 3]
```diff
@@ -116,11 +118,22 @@ public void testFindUserByEmailWithPuiUserManager() {
...
+        roleAdditionResponse.setIdamStatusCode("200");
+        modifyUserRolesResponse.setRoleAdditionResponse(roleAdditionResponse);
+        ObjectMapper mapper = new ObjectMapper();
+        String body = mapper.writeValueAsString(modifyUserRolesResponse);
+        ResponseEntity<Object> responseEntity = ResponseEntity.status(200).body(body);
+        when(professionalUserServiceMock.modifyRolesForUser(any(), any(), any())).thenReturn(responseEntity);
+
         when(userProfileUpdateRequestValidatorMock.validateRequest(userProfileUpdatedData)).thenReturn(userProfileUpdatedData);
 
         String userId = UUID.randomUUID().toString();
```
Comment (KOTLAPRA23) [inline src/test/java/uk/gov/hmcts/reform/professionalapi/controller/internal/ProfessionalUserInternalControllerTest.java:135]: could you please add verify for when conditions?

Comment (shreedhar-hmcts) [reply src/test/java/uk/gov/hmcts/reform/professionalapi/controller/internal/ProfessionalUserInternalControllerTest.java:135]: Done.

File: src/test/java/uk/gov/hmcts/reform/professionalapi/service/impl/ProfessionalUserServiceImplTest.java (hunk 2 above)
Comment (shreedhar-hmcts) [reply src/test/java/uk/gov/hmcts/reform/professionalapi/service/impl/ProfessionalUserServiceImplTest.java:602]: This is not test. A common method used in couple of test cases.

File: src/test/java/uk/gov/hmcts/reform/professionalapi/service/impl/ProfessionalUserServiceImplTest.java (hunk 1 above)
Comment (shreedhar-hmcts) [reply src/test/java/uk/gov/hmcts/reform/professionalapi/service/impl/ProfessionalUserServiceImplTest.java:598]: done.

**Pull request closed without merging**